src/main/java/com/codersbay/gerhofer/Square.java

```diff
@@ -0,0 +1,33 @@
+package com.codersbay.gerhofer;
+
+import java.awt.*;
+
+public class Square extends Rectangle {
+
+    public Square(Color color, boolean isFilled, double length) {
+        super(color, isFilled, length, length);
+    }
+
+    public double getLength() {
+        return super.getLength();
+    }
+
+    public void setLength(double length) {
+        super.setLength(length);
+        super.setWidth(length);
+    }
+
+    public double getArea() {
```

Comment:
as a square is a rectangle no need to define the getArea and getPerimeter methods - you can just use those defined in the rectangle ;)